crates/plugins/src/manifest.rs

```diff
@@ -117,26 +117,51 @@ impl Architecture {
     }
 }
 
-/// Checks whether the plugin supports the currently running version of Spin.
-pub fn check_supported_version(
+/// Checks whether the plugin supports the currently running version of Spin
+/// and prints a warning if not (or if uncertain).
+pub fn warn_unsupported_version(
     manifest: &PluginManifest,
     spin_version: &str,
     override_compatibility_check: bool,
 ) -> Result<()> {
     let supported_on = &manifest.spin_compatibility;
-    inner_check_supported_version(supported_on, spin_version, override_compatibility_check)
+    inner_warn_unsupported_version(supported_on, spin_version, override_compatibility_check)
 }
 
-fn inner_check_supported_version(
-    supported_on: &str,
-    spin_version: &str,
-    override_compatibility_check: bool,
-) -> Result<()> {
+/// Does the manifest compatibility pattern match this version of Spin?  This is a
+/// strict semver check.
+fn is_version_fully_compatible(supported_on: &str, spin_version: &str) -> Result<bool> {
     let comparator = VersionReq::parse(supported_on).with_context(|| {
         format!("Could not parse manifest compatibility version {supported_on} as valid semver")
     })?;
     let version = Version::parse(spin_version)?;
-    if !comparator.matches(&version) {
+    Ok(comparator.matches(&version))
+}
+
+/// This is more liberal than `is_version_fully_compatible`; it relaxes the semver requirement
+/// for Spin pre-releases, so that you don't get *every* plugin showing as incompatible when
+/// you run a pre-release.  This is intended for listing; when executing, we use the interactive
+/// `warn_unsupported version`, which provides the full nuanced feedback.
```

Comment:
```suggestion
/// `warn_unsupported_version`, which provides the full nuanced feedback.
```

Reply:
I'm embarrassed at how long, _even with highlighting_, it took me to spot this.  Thank you!